File: matematyka.rb
module Matematyka
  def self.dodawanie(x,y)
    print x,"+",y,"=",x+y
	print "\n"
  end
  def self.odejmowanie(x,y)
	print x,"-",y,"=",x-y 
	print "\n"
  end
  def self.mnozenie(x,y)
	print x,"*",y,"=",x*y
	print "\n"
  end
  def self.dzielenie(x,y)
    print x,"/",y,"=",x/y
	print "\n"
  end
  def self.potega(x,y)
	result = 1
	n=y
      while y > 0
		result = result * x
		y -= 1
      end
	print x,"^",n,"=",result
	print "\n"
  end
	
end

begin
  print "Podaj co chcesz zrobic:\n1.Dodawanie\n2.Odejmowanie\n3.Mnozenie\n4.Dzielenie\n5.Potegowanie\n6.wyjscie\n"
  print "Wybierz opcje:"
  mode = gets.chomp!
    case mode.to_i
      when 1 then print "\nDodawanie\n"			
        print"\nPodaj 1 wartosc:\n"
        a2=gets.chomp!
        a=a2.to_i
        print"\nPodaj 2 wartosc:\n"
        b2=gets.chomp!
	b=b2.to_i
	Matematyka.dodawanie(a,b)
      when 2 then print "\nOdejmowanie.\n"		
        print"\nPodaj 1 wartosc:\n"
	a2=gets.chomp!
	a=a2.to_i
	print"\nPodaj 2 wartosc:\n"
	b2=gets.chomp!
	b=b2.to_i
	Matematyka.odejmowanie(a,b)
      when 3 then print "\nMnozenie.\n"		
	print"\nPodaj 1 wartosc:\n"
	a2=gets.chomp!
	a=a2.to_i
	print"\nPodaj 2 wartosc:\n"
	b2=gets.chomp!
	b=b2.to_i
	Matematyka.mnozenie(a,b)
      when 4 then print "\nDzielenie.\n"		
	print"\nPodaj 1 wartosc:\n"
	a2=gets.chomp!
	a=a2.to_f
	print"\nPodaj 2 wartosc:\n"
	b2=gets.chomp!
	b=b2.to_f
	Matematyka.dzielenie(a,b)
      when 5 then print "\nPotegowanie.\n"		
	print"\nPodaj 1 wartosc:\n"
	a2=gets.chomp!
	a=a2.to_i
	print"\nPodaj 2 wartosc:\n"
	b2=gets.chomp!
	b=b2.to_i
	Matematyka.potega(a,b)
      when 6 then
	print "\nkoniec programu!\n"
	exit
    end
end

File: matematyka.c
#include <stdio.h>
#include <stdlib.h>

/* run this program using the console pauser or add your own getch, system("pause") or input loop */



int main(int argc, char *argv[]) {
	int wartosc,a,b,wzor,n;
	double x,y,wynik;
	while(wartosc<=5){
	printf ("Podaj co chcesz zrobiæ:\n1.Dodawanie\n2.Odejmowanie\n3.Mnozenie\n4.Dzielenie\n5.Potegowanie\n6.wyjscie\n");
	printf("Podaj opcje:");
	scanf("%i",&wartosc);
	switch (wartosc)
	{
		case 1:
			printf("\nDodawanie\n");
			printf("\nPodaj pierwsza wartosc:");
			scanf("%i",&a);
			printf("\nPodaj druga wartosc:");
			scanf("%i",&b);
			wzor=a+b;
			printf("\n%i+%i=%i\n",a,b,wzor);
		break;
		
		case 2:
			printf("\nOdejmowanie\n");
			printf("\nPodaj pierwsza wartosc:");
			scanf("%i",&a);
			printf("\nPodaj druga wartosc:");
			scanf("%i",&b);
			wzor=a-b;
			printf("\n%i-%i=%i\n",a,b,wzor);
		break;
		
		case 3:
			printf("\nMnozenie\n");
			printf("\nPodaj pierwsza wartosc:");
			scanf("%i",&a);
			printf("\nPodaj druga wartosc:");
			scanf("%i",&b);
			wzor=a*b;
			printf("\n%i*%i=%i\n",a,b,wzor);
		break;
		
		case 4:
			printf("\nOdejmowanie\n");
			printf("\nPodaj pierwsza wartosc:");
			scanf("%lf",&x);
			printf("\nPodaj druga wartosc:");
			scanf("%lf",&y);
			wzor=x/y;
			printf("\n%lf/%lf=%lf\n",x,y,wynik);
		break;
		
		case 5:
			printf("\nPotegowanie\n");
			printf("\nPodaj pierwsza wartosc:");
			scanf("%i",&a);
			printf("\nPodaj druga wartosc:");
			scanf("%i",&b);
			wzor=1;
			n=b;
				while (b > 0){
					wzor = wzor * a;
					b--;
				}
			printf("\n%i^%i=%i\n",a,n,wzor);
		break;
		
		default:
			printf("Koniec programu.");
			break;
	}
}
	return 0;
}
